app/main.py

```diff
@@ -1,2 +1,84 @@
+from copy import deepcopy
+
+from typing import Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self,) -> None:
+
+        self.length = 0
+        self.hash_table = [[] for _ in range(8)]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        threshold = (len(self.hash_table) * 2) // 3
+        if self.length == threshold:
+            self.resize()
+        hash_key = hash(key)
+        index = self.get_index(hash_key)
+
+        if self.hash_table[index]:
+
+            while self.hash_table[index]:
+                if key == self.hash_table[index][0]:
+                    self.hash_table[index][1] = value
+                    return
+                index += 1
+                if index == len(self.hash_table):
+                    index = 0
+            self.hash_table[index] = [key, value, key.__hash__()]
+            self.length += 1
+        else:
+            self.hash_table[index] = [key, value, key.__hash__()]
+            self.length += 1
+
+    def __getitem__(self, key: Any) -> Any:
```

Comment:
You need to use hash func as well to find the item because there is no difference with the list in this solution
You can do the same as in setitem method but instead of rewriting value - return it when the keys are the same

Reply:
This looks better but still has the same logic - you trying to find an item by looking at every element in a list.
Instead, you need to use a hash function to get the index of the element you need to start looking from.